pkg/controller/catalog/server/controller.go

```diff
@@ -52,123 +52,149 @@ func createController(s storage.Storage) (*controller, error) {
 	}, nil
 }
 
-func (c *controller) updateServiceInstance(in *scmodel.ServiceInstance) error {
+func (c *controller) updateServiceInstance(in *servicecatalog.Instance) error {
 	// Currently there's no difference between create / update,
 	// but for prepping for future, split these into two different
 	// methods for now.
 	return c.createServiceInstance(in)
 }
 
-func (c *controller) createServiceInstance(in *scmodel.ServiceInstance) error {
-	broker, err := storage.GetBrokerByServiceClass(c.storage, in.ServiceID)
+func (c *controller) createServiceInstance(in *servicecatalog.Instance) error {
+	broker, err := storage.GetBrokerByServiceClass(c.storage.Brokers(), c.storage.ServiceClasses(), in.Spec.CFServiceID)
```

Comment:
Fixed in https://github.com/kubernetes-incubator/service-catalog/pull/176/commits/ac471a4dd1ef957c19b8e37c3c2611f98e5b4582